File: 0217-contains-duplicate/0217-contains-duplicate.py

#psuedocode

#declare a dictionary to serve as a hash table

#build the hashtable + run the test
#loop over of each num in nums
    #if the num exists in the hashtable as a key -> return true
        #add the number as a key in the hashtable, with a value of one
    
#return false -> does not contain any duplicates
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashmap = {}

        for num in nums:
            if num in hashmap:
                return True
            else:
                hashmap[num] = 1
        
        return False
        

File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */

/**

freq table problem

create a freq table

iterate over nums
for each num in nums

if the num is already in nums, return false, otherwise map[nums] = true

if the above completes, return true

*/

var containsDuplicate = function(nums) {
    let map = {}

    for (const num of nums) {
        if (num in map) return true
        map[num] = true
    }
    
    return false
};